File: python/excecoes.py
class PilhaCheiaErro(Exception):
    """Levantada ao tentar empilhar em uma pilha que atingiu a capacidade máxima."""
    pass

class PilhaVaziaErro(Exception):
    """Levantada ao tentar desempilhar ou acessar elementos de uma pilha vazia."""
    pass

class TipoErro(TypeError):
    """Levantada ao tentar empilhar um dado de tipo incompatível com a pilha."""
    pass

File: js/Excecoes.js
class PilhaCheiaErro extends Error {
    constructor(message = "Pilha Cheia") {
        super(message);
        this.name = "PilhaCheiaErro";
    }
}

class PilhaVaziaErro extends Error {
    constructor(message = "Pilha Vazia") {
        super(message);
        this.name = "PilhaVaziaErro";
    }
}

class TipoErro extends TypeError {
    constructor(message = "Tipo Incompatível") {
        super(message);
        this.name = "TipoErro";
    }
}

module.exports = { PilhaCheiaErro, PilhaVaziaErro, TipoErro };